lib/rules/base.js

```diff
@@ -448,7 +448,10 @@ module.exports = class Base {
       }
 
       try {
+        const { determineRuleConfig } = require('../get-config');
```

Comment:
Lets hoist this require to the top.

Reply:
This is due to the circular dependency of modules that I had to load it dynamically. As discussed, I ll pull this function out and other simpler ones  from `get-config.js` in another PR to be in a separate util file to resolve the circular dependency issues